projects/canopy/src/lib/forms/sort-code/sort-code.component.ts

```diff
@@ -0,0 +1,206 @@
+  third: FormControl;
+  valueChanges: Subscription;
+  hasError: boolean;
+
+  @Input() value: string;
+  @Input() disabled: false;
+  @Input() firstId = `lg-input-sort-code-first-${this.uniqueId++}`;
+  @Input() secondId = `lg-input-sort-code-second-${this.uniqueId++}`;
+  @Input() thirdId = `lg-input-sort-code-third-${this.uniqueId++}`;
+  @Input() ariaDescribedBy: string;
+
+  _hintElement: LgHintComponent;
+  @ContentChild(LgHintComponent)
+  set hintElement(element: LgHintComponent) {
+    this.ariaDescribedBy = this.domService.toggleIdInStringProperty(
+      this.ariaDescribedBy,
+      this._hintElement,
+      element,
+    );
+    this._hintElement = element;
+  }
+
+  _validationElement: LgValidationComponent;
+  @ContentChild(LgValidationComponent)
+  set errorContentElement(element: LgValidationComponent) {
+    this.ariaDescribedBy = this.domService.toggleIdInStringProperty(
+      this.ariaDescribedBy,
+      this._validationElement,
+      element,
+    );
+    this._validationElement = element;
+  }
+
+  @ViewChild('sortCodeFormDirective')
+  formGroupDirective: FormGroupDirective;
+
+  constructor(
+    private domService: LgDomService,
+    private errorState: LgErrorStateMatcher,
+    @Self()
+    @Optional()
+    private ngControl: NgControl,
+    private parentFormGroupDirective: FormGroupDirective,
+  ) {
+    if (this.ngControl != null) {
+      this.ngControl.valueAccessor = this;
+    }
+    this.first = new FormControl(null, [
+      Validators.required,
+      Validators.pattern(/^\d\d$/),
+    ]);
+    this.second = new FormControl(null, [
+      Validators.required,
+      Validators.pattern(/^\d\d$/),
+    ]);
+    this.third = new FormControl(null, [
+      Validators.required,
+      Validators.pattern(/^\d\d$/),
+    ]);
+    this.sortCodeFormGroup = new FormGroup(
+      {
+        first: this.first,
+        second: this.second,
+        third: this.third,
+      },
+      {
+        updateOn: 'blur',
+      },
+    );
+  }
+
+  ngOnInit(): void {
+    // append internal validators to external validators if applicable.
+    if (this.ngControl && this.ngControl.control) {
+      this.ngControl.control.setValidators([this.validate.bind(this)]);
+      this.ngControl.control.updateValueAndValidity();
+    }
+
+    this.valueChanges = this.sortCodeFormGroup.valueChanges.subscribe(
+      (sortCode: SortCodeField) => {
+        const first = sortCode.first || '';
+        const second = sortCode.second || '';
+        const third = sortCode.third || '';
+        this.onChange(`${first}${second}${third}`);
+      },
+    );
+
+    // submit the group when the parent form is submitted
+    this.parentFormGroupDirective.ngSubmit
+      .pipe(filter(({ type }) => type === 'submit'))
+      .subscribe((event) => {
+        this.formGroupDirective.onSubmit(event);
+        this.ngControl.control.updateValueAndValidity();
+      });
+  }
+
+  isControlInvalid(control: NgControl, form: FormGroupDirective) {
```

Comment:
This should probably have a return type of `boolean` (I think).

Reply:
Yes I think so too

Reply:
Done ✅